torch/_inductor/codecache.py

```diff
@@ -1308,6 +1308,27 @@ def result(self):
         return kernel
 
 
+# If this process dies abnormally (e.g. segfault)
+# it will not shut down the workers. Instead
+# the workers will have their parent reassigned to the
+# init process. This launches a separate thread to
+# watch for the worker getting reassigned,
+# and cleans it up in this case.
+#
+# This function cannot be an inner function since otherwise mp_context="spawn" would
+# not work for ProcessPoolExecutor since inner functions cannot be pickled.
+def _async_compile_initializer(orig_ppid):
```

Comment:
We should use safe abstractions for process control like pidfs to avoid race conditions.

Reply:
preexisting condition